libs/cart/src/effects/cart-resolver.effects.spec.ts

```diff
@@ -56,161 +59,285 @@ describe('DaffCartResolverEffects', () => {
+        });
+
+        expect(effects.onResolveCart$).toBeObservable(expected);
+      });
+    });
+
+    describe('handling XHR errors when retrieving a cart', () => {
+      it('should dispatch a DaffCartLoadFailure action', () => {
+        const response = cold('#', {});
+        driver.get.and.returnValue(response);
+        driver.create.and.returnValue(response);
+
+        const loadCartFailureAction = new DaffCartLoadFailure(
+          'Cart loading has failed',
+        );
+
+        actions$ = hot('--a', { a: new DaffResolveCart() });
+        const expected = cold('--b', {
+          b: loadCartFailureAction
+        });
+
+        expect(effects.onResolveCart$).toBeObservable(expected);
+      });
+    });
+
+    describe('creating a cart when there is no cart id in storage', () => {
+      it('should create a cart', () => {
+        cartStorageService.getCartId.and.returnValue(undefined);
+        driver.create.and.returnValue(of({ id: stubCart.id }));
+        driver.get.and.returnValue(of(stubCart));
+
+        const loadCartSuccessAction = new DaffCartLoadSuccess(stubCart);
+
+        actions$ = hot('--a', { a: new DaffResolveCart() });
+        const expected = cold('--b', {
+          b: loadCartSuccessAction
+        });
+
+        expect(effects.onResolveCart$).toBeObservable(expected);
+        expect(driver.create).toHaveBeenCalled();
+      });
+
+      it('should set the cart id in local storage', () => {
+        cartStorageService.getCartId.and.returnValue(undefined);
+        driver.create.and.returnValue(of({ id: stubCart.id }));
+        driver.get.and.returnValue(of(stubCart));
+
+        const loadCartSuccessAction = new DaffCartLoadSuccess(stubCart);
+
+        actions$ = hot('--a', { a: new DaffResolveCart() });
+        const expected = cold('--b', {
+          b: loadCartSuccessAction
+        });
+
+        expect(effects.onResolveCart$).toBeObservable(expected);
+        expect(cartStorageService.setCartId).toHaveBeenCalledWith(String(stubCart.id));
+      });
+    });
+
+    describe('when the error thrown is a DaffCartNotFoundError', () => {
+
+      it('should create a new cart', () => {
+        cartStorageService.getCartId.and.returnValue('id');
+        const response = cold('#', {}, new DaffCartNotFoundError('error'));
+        driver.get.and.returnValue(response);
+
+        const cartCreateAction = new DaffCartCreate();
+
+        actions$ = hot('--a', { a: new DaffResolveCart() });
+        const expected = cold('--b', {
+          b: cartCreateAction,
+        });
+
+        expect(effects.onResolveCart$).toBeObservable(expected);
+      });
+    });
+
+    describe('when the error thrown is a DaffStorageServiceError', () => {
+
+      it('should indicate that the storage service has failed', () => {
+        const response = cold('#', {}, new DaffStorageServiceError());
+        driver.get.and.returnValue(response);
+        driver.create.and.returnValue(response);
```

Comment:
These services shouldn't throw this error, right?

Reply:
I didn't write any of these tests but you're right, this test is set up incorrectly. fixed

Reply:
Yea, I think I did. Oops